raiden/transfer/architecture.py

```diff
@@ -264,6 +280,14 @@ def dispatch(self, state_change: StateChange) -> List[Event]:
 
         assert isinstance(iteration, TransitionResult)
 
+        # if any balance proofs have changed, create events
+        balance_proofs = self.detect_balance_proof_change(
```

Comment:
I don't think this needs to be part of the state machine, this code is not important for the protocol or restartability, so it can safely be called from somewhere else

Reply:
where would you rather see this integrated?

Reply:
sorry, I missed this question: https://github.com/raiden-network/raiden/pull/3452#discussion_r255958278